src/customParser.ts

```diff
@@ -0,0 +1,330 @@
+import BoxGroup from './geometry/BoxGroup';
+import CircleGroup from './geometry/CircleGroup';
+import ConeGroup from './geometry/ConeGroup';
+import * as THREE from 'three';
+
+const lookup = [1, 2];
+for (let i = 2; i <= 77; i++) {
+  lookup.push(lookup[i - 1] + lookup[i - 2]);
+}
+
+class FibonacciDecoder {
+  public numberRead: number;
+
+  private buffer: ArrayBuffer;
+  private data: Uint8Array;
+
+  private readBitId: number;
+  private currentValue: number;
+  private nextFibIndex: number;
+
+  constructor(compressedBuffer: ArrayBuffer, startByte?: number, length?: number) {
+    if (!startByte) { startByte = 0; }
+    if (!length) { length = compressedBuffer.byteLength - startByte; }
+
+    this.numberRead = 0;
+    this.buffer = compressedBuffer;
+    this.data = new Uint8Array(this.buffer, startByte, length);
+
+    // read first bit
+    this.currentValue = (this.data[0] & (1 << 7)) >> 7;
+    this.nextFibIndex = 1;
+    this.readBitId = 1;
+  }
+
+  nextValue() {
+    this.numberRead++;
+    // read rest of file
+    for (let i = 0; i < 100; i++) {
```

Comment:
What's this?

Reply:
Function that reads one additional value

Reply:
Why 100? Before it said `while (true)`. Seems very arbitrary with a for loop to 100 :P